### DISCUSSION TITLE
GraphQL API: How to fetch all repositories that the current user has access to

### DISCUSSION THREAD

**Discussion by jumarko** [community/community#24860] [Apps, API and Webhooks]
<p>I’m trying to find a GraphQL equivalent of the <code>/user/repos</code> resource: <a href="https://developer.github.com/v3/repos/#list-your-repositories" rel="nofollow noopener">https://developer.github.com/v3/repos/#list-your-repositories</a></p>
<p>&gt; List repositories that the authenticated user has explicit permission (:read, :write, or :admin) to access.</p>
<p>I’ve tried following query but that only returns repositories under my account, and not the other repositories I have access to (organizational repos - both public and private):</p>
<pre><code>{
  viewer {
    repositories(first: 100, after: properCursorHereOrOmitIfFirstPage, orderBy: {field:NAME, direction:ASC}) {
      pageInfo {hasNextPage, endCursor}
      nodes {
        name
        url
        isPrivate
        owner {
          login
        }
        defaultBranchRef {
          name
        }
      }
    }
  }
}
</code></pre>
<p>EDIT: </p>
<p>I’ve also tried to use affiliations but without success  - still returns only repositories directly under my “jumarko” account:</p>
<pre><code>{
  viewer {
    repositories(first: 100, affiliations: [OWNER, COLLABORATOR, ORGANIZATION_MEMBER]) {
      totalCount
      pageInfo {
        endCursor
        hasNextPage
      }
      nodes{
        name
          owner {
            login
          }
        }
      }
   }
 }
</code></pre>

**Comment by ahaessly** [comment]
<p>try repositoriesContributedTo instead of repositories</p>

**Comment by jumarko** [comment]
<p>Thanks for the suggestion but that doesn’t sound right.</p>
<p>I want to get a list of all repositories I have access to, not only the ones I’ve contributed too.</p>
<p>As mentioned before, I want an equivalent of _user/_repos endpoint of v3 API, which for more returns 2300+ repositories while the “equivalent” GraphQL API query (see my previous post) returns only 207 repositories.</p>
<p><em>repositoriesContributedTo</em> I contributed to returns only 85 repos (with includeUserRepositories set to true).</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hey <a class="mention" href="/jumarko">@jumarko</a>,</p>
<p>Thanks for being here, have you tried? </p>
<pre><code>{
  viewer {
    repositories(first: 100) {
      totalCount
      nodes {
        nameWithOwner
      }
      pageInfo {
        endCursor
        hasNextPage
      }
    }
  }
}
</code></pre>

**Comment by jumarko** [comment]
<p><a class="mention" href="/andreagriffiths11">@andreagriffiths11</a> thanks for the suggestion but that does the same thing as my first query.</p>

**Comment by machour** [comment]
<p>Seems to be already reported and awaiting for GitHub feedback: <a href="https://github.community/t/grapql-viewer-repositories-gets-only-repositories-where-the-user-is-collaborator/13579">https://github.community/t5/GitHub-API-Development-and/Grapql-viewer-repositories-gets-only-repositories-where-the-user/m-p/14318?collapse_discussion=true&amp;page=2&amp;q=graphql%20user%20repositories&amp;search_type=thread</a></p>

**Comment by krasimir** [comment]
<p><a class="mention" href="/jumarko">@jumarko</a> Did you find a way to get all the repos. I really don’t wanna use v3 for this.</p>

**Comment by jumarko** [comment]
<p><a class="mention" href="/krasimir">@krasimir</a> No, I use V3 rest api to load list of repos.</p>

**Comment by krasimir** [comment]
<p>Ok, fair enough. I’ll do the same for the time being.</p>

**Comment by insidewhy** [comment, marked as answer]
<p>I found the solution!</p>
<p>Add the following parameters to your <code>repositories</code> query: <code>affiliations:[OWNER, ORGANIZATION_MEMBER, COLLABORATOR],&amp;nbsp;ownerAffiliations:[OWNER, ORGANIZATION_MEMBER, COLLABORATOR]</code></p>
<p>Not sure why this is necessary to retrieve private organisation repos and not public organisation repos, but nevertheless it works.</p>
